### DISCUSSION TITLE
How to backtrace

### DISCUSSION THREAD

**Discussion by SamueleD78** [vfsfitvnm/frida-il2cpp-bridge#161] [Q&A]
Fist, and always, thank you for this wonderful tool!

I've tried to backtrace, inside an implementation hook, and what i receive is a list of addresses:

```
func.implementation = function (): Il2Cpp.String {
    console.log(Thread.backtrace());
    return this.method<Il2Cpp.String>("func_name").invoke();
};

```

`0x116c53294,0x116c912c4,0x116c9a024,0x116c9af2c,0x116c99d2c,0x116c912c4,0x116c9a024,0x116c99d2c,0x116caca04,0x116c9b08c,0x116c99d2c,0x116c47af8,0x116c4d5ac,0x116b64cf8,0x116b681b4,0x10916a428
`

How can i tralsate them to objects/functions?

**Comment by vfsfitvnm** [comment]
Hi, you can read issue https://github.com/vfsfitvnm/frida-il2cpp-bridge/issues/43

**Reply by SamueleD78** [reply]
i'm sorry, i've search for it and honestly i missed it. I'm really sorry.

Anyway i've tried it, and the result is:

```
0x10916a428 UnityFramework!0x270e428 (0x270e428)
0x108d8be08 UnityFramework!0x232fe08 (0x232fe08)
0x109162cb4 UnityFramework!0x2706cb4 (0x2706cb4)
0x109dfe60c UnityFramework!0x33a260c (0x33a260c)
0x1093d3d5c UnityFramework!0x2977d5c (0x2977d5c)
0x10a2e7088 UnityFramework!0x388b088 (0x388b088)
0x109cff560 UnityFramework!0x32a3560 (0x32a3560)
0x109e6d52c UnityFramework!0x341152c (0x341152c)
0x109607210 UnityFramework!0x2bab210 (0x2bab210)
0x106c80884 UnityFramework!0x224884 (0x224884)
0x107cc4df8 UnityFramework!0x1268df8 (0x1268df8)
0x10744645c UnityFramework!0x9ea45c (0x9ea45c)
0x1074523a4 UnityFramework!0x9f63a4 (0x9f63a4)
0x10745cb4c UnityFramework!0xa00b4c (0xa00b4c)
0x107257abc UnityFramework!0x7fbabc (0x7fbabc)
0x10735d808 UnityFramework!0x901808 (0x901808)

```
something better, but still not understandable XD

**Reply by vfsfitvnm** [reply]
Yes, this is the best thing you can get without using a disassembler

**Reply by SamueleD78** [reply]
Ok last question: in the wiki it's written to do Il2Cpp.dump().methods().build()
but this don't work anymore, methods it's not found.
What is the correct way to do it?

**Reply by vfsfitvnm** [reply]
That one was removed because not so much useful

**Reply by SamueleD78** [reply]
ok but... what is the correct way then?

**Reply by vfsfitvnm** [reply]
To do what? Just perform a normal dump
